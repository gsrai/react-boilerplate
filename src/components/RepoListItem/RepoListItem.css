.repo-list-item {
  padding: 10px;
  margin: 10;
  background: 'white';
  box-shadow: 0 1px 5px rgba(0,0,0,0.5);
}