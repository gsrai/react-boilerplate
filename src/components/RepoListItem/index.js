import React from 'react'
import PropTypes from 'prop-types'

import './RepoListItem.css'

const RepoListItem = ({ repo }) => {
  return (
    <div className={'repo-list-item'}>
      <div>
        <a href={repo.html_url} target={'_blank'}>
          { repo.full_name }
        </a>
        ðŸŒŸ<strong>{ repo.stargazers_count }</strong>
      </div>
      <p>{ repo.description }</p>
    </div>
  )
}

RepoListItem.propTypes = {
  repo: PropTypes.object
}

export default RepoListItem
