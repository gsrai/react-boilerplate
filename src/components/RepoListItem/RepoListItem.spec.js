import { shallow } from 'enzyme'
import React from 'react'
import RepoListItem from './index'

const REPO_MOCK = {
  full_name: 'facebook/react',
  html_url: 'https://github.com/facebook/react',
  description: 'A declarative, efficient, and flexible JavaScript library for building user interfaces.',
  stargazers_count: 108963
}

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(<RepoListItem repo={REPO_MOCK} />)
    expect(wrapper).toMatchSnapshot()
  })
})
