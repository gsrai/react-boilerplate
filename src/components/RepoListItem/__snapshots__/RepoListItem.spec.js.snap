// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div
  className="repo-list-item"
>
  <div>
    <a
      href="https://github.com/facebook/react"
      target="_blank"
    >
      facebook/react
    </a>
    ðŸŒŸ
    <strong>
      108963
    </strong>
  </div>
  <p>
    A declarative, efficient, and flexible JavaScript library for building user interfaces.
  </p>
</div>
`;
